### DISCUSSION TITLE
Adonisjs mail with sendgrid API Key

### DISCUSSION THREAD

**Discussion by farnetani** [adonisjs/core#1898] [Help]
Sendgrid is changing its authentication rules and will stop working on the 15th now to accept simple confirmation with login and password, it will only work with API KEY, does anyone here know how to change this in my Adonisjs 4.1 so that I can continue to use sendgrid as smtp?

**Comment by junicodes** [comment]
>     I think you should replace the username field with a a simple text
apikey and then your password section should be replace with actual API key
from sendgrid itself ...try this and let me know.

> MAIL_USERNAME=apikey
>
> MAIL_PASSWORD=your_sendgrid_api_key
>
>

**Comment by ghost** [comment]
will SendGrid API key and 2FA migration will work with the same account even with base authentication also until the deadline date.  once i migration what are the changes nee to be done in code to send emails as SMTP after 2FA Migation.
